osidb/serializer.py

```diff
@@ -1615,6 +1616,36 @@ class Meta:
             + TrackingMixinSerializer.Meta.fields
         )
 
+    def create(self, validated_data):
+        """Handles reference creation and sync with Jira trackers if needed."""
+        new_ref = super().create(validated_data)
+        self.update_trackers(None, new_ref)
+        return new_ref
+
+    def update(self, instance, validated_data):
+        """Handles reference update and sync with Jira trackers if needed."""
+        old_ref = FlawReference.objects.get(uuid=instance.uuid)
+        new_ref = super().update(instance, validated_data)
+        self.update_trackers(old_ref, new_ref)
+        return new_ref
+
+    def update_trackers(self, old_ref, new_ref):
+        """
+        Updates the related Jira trackers passing the references as links if needed.
+        """
+        if old_ref is not None and not differ(old_ref, new_ref, ["url", "description"]):
+            return
+
+        for affect in new_ref.flaw.affects.all():
+            if affect.is_community:
+                continue
+
+            for tracker in affect.trackers.all():
+                if not tracker.is_closed and tracker.type == Tracker.TrackerType.JIRA:
+                    tracker.sync_reference(
+                        jira_token=self.get_jira_token(), reference=new_ref
+                    )
```

Comment:
I do not want to over-complicate this right now but I am not convinced that the `serializer` module is a good place for this on-save logic and we should rather continuously migrate it to the model layer. I know that we already backed a lot of it into the `serializer` for various reasons in the past and it is not always trivial to change so I am just leaving a comment and not insisting on the change.